src/python/pants/engine/mapper.py

```diff
@@ -140,17 +140,21 @@ def create(cls, spec_path, address_maps):
                                            current_path=current_path))
         objects_by_name[name] = (current_path, obj)
     return AddressFamily(namespace=spec_path,
-                         objects_by_name=OrderedDict((name, obj) for name, (_, obj)
-                                                     in sorted(objects_by_name.items())))
+                         objects_by_name=OrderedDict((name, (path, obj)) for name, (path, obj)
+                                                      in sorted(objects_by_name.items())))
 
   @memoized_property
   def addressables(self):
-    """Return a mapping from address to thin addressable objects in this namespace.
+    """Return a mapping from BuildFileAddress to thin addressable objects in this namespace.
 
-    :rtype: dict from :class:`pants.build_graph.address.Address` to thin addressable objects.
+    :rtype: dict from :class:`pants.build_graph.address.BuildFileAddress` to thin addressable
+            objects.
     """
-    return {Address(spec_path=self.namespace, target_name=name): obj
-            for name, obj in self.objects_by_name.items()}
+    addressables = {}
+    for (name, value) in self.objects_by_name.items():
+      path, obj = value
```

Comment:
should be able to consolidate the tuple unpacking here like:

```
for name, (path, obj) in self.objects_by_name.items():
  ...
```

and alternatively, keep this as a dict comp (which might be faster):

```
return {
  BuildFileAddress(rel_path=path, target_name=name): obj
  for name, (path, obj) in self._objects_by_name.items()
}
```